src/aiortc/contrib/media.py

```diff
@@ -79,6 +80,89 @@ async def stop(self):
         self.__tracks = {}
 
 
+class BroadcasterStreamTrack(MediaStreamTrack):
+    def __init__(self, broadcaster, source: MediaStreamTrack) -> None:
+        super().__init__()
+        self.kind = source.kind
+        self._broadcaster = broadcaster
+        self._queue: asyncio.Queue[Optional[Frame]] = asyncio.Queue()
```

Comment:
In case of using this track for transmitting a camera feed, it would be useful only send the latest frame and not store the complete queue.

However I could image in the case when you use it at the receiving side, you want to receive all incoming frames

Reply:
There is a bit of a discreptancy here, because we have two different behaviours:

- for the first subscriber we only start queueing frames once recv() has been called
- for subsequent subscribers, we start queuing frames as soon as subscribe() has been called. This means during the time between subscribe() and the first call to recv() (typically the time required to establish ICE / DTLS) we are building up a backlog.

Reply:
Building up a backlog on transceiving is indeed undesired.

Building up a backlog (after established ICE / DTLS, in other words: not the startup backlog) on receiving in BroadcasterStreamTrack could be desired behaviour.

How about registering all proxy queues to the source on first call to recv()?

Reply:
Yup, this should be fixed in d20343e5128dd7aff651ba8c4545d8104525cf5c